src/Composer/Command/SelfUpdateCommand.php

```diff
@@ -156,6 +157,15 @@ class_exists('Composer\Downloader\FilesystemException');
             return $this->rollback($output, $rollbackDir, $localFilename);
         }
 
+        if ($input->getArgument('command') === 'self' && $input->getArgument('version') === 'update') {
+            $question = '<info>"update"</info> is not a valid version. Did you mean "composer self-update" [<comment>Y/n</comment>]? ';
+            if ($io->isInteractive() && $io->askConfirmation($question, false)) {
+                $input->setArgument('version', null);
+            } else {
+                throw new CommandNotFoundException(sprintf('The command "%s" does not exist.', $input->getArgument('command')));
+            }
```

Comment:
I think we could just always fix this silently, it's pretty obvious what the reason is, and there's no point in asking for confirmation nor in failing hard if non-interactive.

```suggestion
            $input->setArgument('version', null);
```

Reply:
Sure, this one's updated